### PR TITLE
meson: add explicit BR's & drop un-needed BR's to enable ptest

### PR DESCRIPTION
**Pull request by mfrw**
Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Fix ptest build. All tests pass!

###### Change Log  

- Drop BR on `gmock-devel` & `gtest-devel`
- Add an explicit BR on `libgcrypt-devel, glib, gtest, gmock, cmake & git` to enable ptest

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- NA

###### Links to CVEs  
- NA

###### Test Methodology

- Pipeline build id: Local Build with RUN_CHECK=y

### CODE REVIEW

File: SPECS/meson/meson.spec [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
 Source0:        https://github.com/mesonbuild/meson/releases/download/%{version}/%{name}-%{version}.tar.gz
 BuildRequires:  gcc
 BuildRequires:  gettext
-BuildRequires:  gmock-devel
-BuildRequires:  gtest-devel
 BuildRequires:  ninja-build
 BuildRequires:  python3-devel
 BuildRequires:  python3-libs
 BuildRequires:  python3-setuptools
 BuildRequires:  python3-xml
```
Comment (thcrain-msft) [inline SPECS/meson/meson.spec:16]: Nit: `python3-xml` can be removed as a BR. It has been folded into the `python3-libs` package in 2.0.

**Review by thcrain-msft — APPROVED** [review]
Change looks good to me! Just a couple of `python3` BR cleanup requests for you.

File: SPECS/meson/meson.spec [hunk 2]
```diff
@@ -1,21 +1,28 @@
 Summary:        Extremely fast and user friendly build system
 Name:           meson
 Version:        0.60.2
-Release:        1%{?dist}
+Release:        2%{?dist}
```
Comment (thcrain-msft) [inline SPECS/meson/meson.spec:4]: Looks like you need to bump the toolchain manifest for this package, by the way.

File: SPECS/meson/meson.spec (hunk 1 above)
Comment (mfrw) [reply SPECS/meson/meson.spec:16]: I did not know that. Thanks :)

**Review by PawelWMS — APPROVED** [review]
Same comment as @thcrain-msft: looks good, just need an update to the manifest files.

**Pull request merged**